preprocessor/src/main/java/com/azure/autorest/preprocessor/Preprocessor.java

```diff
@@ -152,9 +156,13 @@ public static CodeModel convertOptionalConstantsToEnum(CodeModel codeModel) {
                 o.getParameters().stream().filter(p -> !p.isRequired() && p.getSchema() instanceof ConstantSchema)
                     .forEach(p -> {
                         ConstantSchema constantSchema = (ConstantSchema) p.getSchema();
-                        SealedChoiceSchema sealedChoiceSchema = convertedChoiceSchemas.computeIfAbsent(constantSchema,
-                            Preprocessor::convertToChoiceSchema);
-                        p.setSchema(sealedChoiceSchema);
+                        if (schemaIsConstantWithChoice.apply(constantSchema)) {
+                            p.setSchema(constantSchema.getValueType());
+                        } else {
+                            SealedChoiceSchema sealedChoiceSchema = convertedChoiceSchemas.computeIfAbsent(constantSchema,
+                                    Preprocessor::convertToChoiceSchema);
+                            p.setSchema(sealedChoiceSchema);
+                        }
```

Comment:
When `valueType` already an enum (open or closed), use that enum. Do not create another enum.

Reply:
And this way, the API be more consistent to the `required` case (`colorValue` or `colorModelValue`).

Reply:
I think Swagger won't have such case.